### ISSUE TITLE
List literals

### ISSUE THREAD

**Issue by rexim** [morganey-lang/Morganey#66]
Derived from #58 

Not sure what's the best syntax for that, but maybe square brackets are ok: `[1, 2, 3]`

**Comment by keddelzz** [comment]
Do you mind if I implement this issue, once branch [62-empty-list](https://github.com/rexim/Morganey/tree/62-empty-list) is merged? Which syntax do we want for lists?

I have to confess I don't like list-literals, whose elements are separated by spaces. Do you insist having the syntax proposed in the description of this issue, or could we discuss using `,` as a separator?

**Comment by rexim** [comment]
Oh my! I'm really sorry! Didn't know I block you! I was planning to wrap up with #62 on Saturday, but I'll try to find some time on Wednesday ('cause today I'm preparing for the Haskell stream).

Of course, I don't mind if you take this! :) Comma as a separator is good. I don't remember why I separated elements with spaces only. Probably was just rush typing. Let me fix the description...

**Comment by keddelzz** [comment]
You don't block me :) Take your time, there is no need to hurry.

If I really wanted I could implement this now and add support for empty lists (`[]`) after #62 is done.

**Comment by keddelzz** [comment]
Should we also add haskells `[1,2 .. 10]` notation for ints and chars?

**Comment by rexim** [comment]
> Should we also add haskells `[1,2 .. 10]` notation for ints and chars?

Would be good to implement. But it sounds like a separate issue to me.

**Comment by keddelzz** [comment]
> But it sounds like a separate issue to me.

I disagree! I think they feel also like list-literals in a sense. So why not add this to the scope of this issue?

**Comment by rexim** [comment]
@keddelzz, lol, that's just my opinion. :) You asked for it, I gave it to you. If you think that I'm gonna decline a PR just because it contains a small thing (which should not by the way screw up the entire semantic of the language) that I believe is slightly out-of-scope, well that's not true. :D Wanna do that in the scope of this issue? Sure, why not. Just make a PR and we gonna discuss everything there. You can make a PR without any issue at all. Technically on GitHub PR is an issue itself. You can propose a feature and its implementation simultaneously with such mechanism. I'm pedantic sometimes not because I like to torture people. I have a really bad memory and I have to deal with thousands different things everyday. And Morganey is just one of them. That's why I like small fine grained issues. But you can propose whatever you want. Just make a PR and we will discuss it.

**Comment by keddelzz** [comment]
Ok :+1:.

## 

To sum our conversation up: I'ill implement this issue after #62 is done and add the notation `[1,2 .. 10]` as well.
